dev-db/clickhouse/clickhouse-9999.ebuild

```diff
@@ -0,0 +1,724 @@
+		"USE_RAPIDJSON \
+			$(usex json "$(usex '!cpu_flags_x86_avx2' json '!cpu_flags_x86_avx2')" json)"
+		"USE_SIMDJSON \
+			$(usex json "$(usex cpu_flags_x86_avx2 json cpu_flags_x86_avx2)" json)"
+	)
+
+	for line in "${CLICKHOUSE_CMAKE_CONFIG_CHECKS[@]}"; do
+		read -r cmake_var use_flag <<< "${line}"
+		echo "if ($(
+			[[ "${use_flag}" ]] && usex "${use_flag}" "NOT" "" || echo NOT
+		) ${cmake_var})" >> CMakeLists.txt
+		echo "    message (FATAL_ERROR \"Failed to $(
+			[[ "${use_flag}" ]] && usex "${use_flag}" use disable || echo check
+		) ${use_flag}: ${cmake_var}\")" >> CMakeLists.txt
+		echo "endif ()" >> CMakeLists.txt
+	done
+}
+
+src_configure() {
+	use cpu_flags_x86_sse4_1 && append-cflags "-msse4.1" && append-cxxflags "-msse4.1"
+	use cpu_flags_x86_sse4_2 && append-cflags "-msse4.2" && append-cxxflags "-msse4.2"
+	use cpu_flags_x86_ssse3 && append-cflags "-mssse3" && append-cxxflags "-mssse3"
+	use cpu_flags_x86_avx && append-cflags "-mavx" && append-cxxflags "-mavx"
+	use cpu_flags_x86_avx2 && append-cflags "-mavx2" && append-cxxflags "-mavx2"
+	use cpu_flags_x86_popcnt && append-cflags "-mpopcnt" && append-cxxflags "-mpopcnt"
+
+	if use ppc64; then
+		 append-cflags "-maltivec" "-D__SSE2__=1" "-DNO_WARN_X86_INTRINSICS"
+		 append-cxxflags "-maltivec" "-D__SSE2__=1" "-DNO_WARN_X86_INTRINSICS"
+	fi
+
+	local line mycmakeargs use_flag cmake_option cmake_var
+	mycmakeargs=(
+		-D"CMAKE_INTERPROCEDURAL_OPTIMIZATION"="$(usex lto ON OFF)"
+		-D"DISABLE_CPU_OPTIMIZE"="ON"
+		-D"HAVE_SSE41"="$(usex cpu_flags_x86_sse4_1)"
+		-D"HAVE_SSE42"="$(usex cpu_flags_x86_sse4_2)"
+		-D"HAVE_SSSE3"="$(usex cpu_flags_x86_ssse3)"
+		-D"HAVE_AVX"="$(usex cpu_flags_x86_avx)"
+		-D"HAVE_AVX2"="$(usex cpu_flags_x86_avx2)"
+		-D"HAVE_POPCNT"="$(usex cpu_flags_x86_popcnt)"
+	)
+
+	for line in "${CLICKHOUSE_CMAKE_USE_ARGUMENTS[@]}"; do
+		read -r cmake_option use_flag <<< "${line}"
+		mycmakeargs+=( "-D${cmake_option}=$(
+			[[ ${use_flag} ]] && usex "${use_flag}" ON OFF || echo ON
+		)" )
+	done
+
+	mycmakeargs+=(
+		-D"BUILD_SHARED_LIBS"="$(usex static-libs OFF ON)"
+		-D"USE_STATIC_LIBRARIES"="$(usex static ON OFF)"
+		-D"MAKE_STATIC_LIBRARIES"="$(usex static-libs ON OFF)"
+		-D"CLICKHOUSE_SPLIT_BINARY"="$(usex split-binary ON OFF)"
+
+		-D"ENABLE_IPO"="$(usex lto ON OFF)"
+
+		-D"ENABLE_TESTS"="$(usex test ON OFF)"
+		-D"ENABLE_CODE_QUALITY"="$(usex test ON OFF)"
+		-D"WITH_COVERAGE"="$(usex test ON OFF)"
+		-D"CMAKE_DISABLE_FIND_PACKAGE_GTest"="$(usex test OFF ON)"
+
+		-D"ENABLE_CLICKHOUSE_ALL"="$(usex tools ON OFF)"
+		-D"ENABLE_CLICKHOUSE_SERVER"="$(usex server ON OFF)"
+		-D"ENABLE_CLICKHOUSE_CLIENT"="$(usex client ON OFF)"
+		-D"ENABLE_CLICKHOUSE_ODBC_BRIDGE"="$(usex odbc "$(usex tools ON OFF)" OFF)"
+		-D"ENABLE_UTILS"="$(usex tools ON OFF)"
+
+		# setting it to ON breaks configuration process because
+		# ClickHouse is not trying to set up link targets and other
+		# compile options even when this is ON
```

Comment:
Did you report a bug upstream?

Reply:
They don't consider unbounded non-static versions as officially supported. So occasionally they fix this things, but because of unofficial status it is more like a chronic disease, so it is easier to deal with each component individually

Reply:
One possible effect of changing this option would be that new features requiring new submodules from 9999 build will be turned off by default. So I think this is the best solution we have. Please resolve the thread